x-pack/plugins/file_upload/public/components/json_upload_and_parse.tsx

```diff
@@ -52,53 +59,43 @@ export class JsonUploadAndParse extends Component {
     this._isMounted = false;
     if (this._geojsonImporter) {
       this._geojsonImporter.destroy();
-      this._geojsonImporter = null;
+      this._geojsonImporter = undefined;
     }
   }
 
   componentDidUpdate() {
-    this._setIndexReady();
     if (this.props.isIndexingTriggered && this.state.phase === PHASE.CONFIGURE) {
       this._import();
     }
   }
 
-  _setIndexReady = () => {
-    const isIndexReady =
-      this._geojsonImporter !== undefined &&
-      !!this.state.selectedIndexType &&
-      !!this.state.indexName &&
-      !this.state.hasIndexErrors &&
-      this.state.phase === PHASE.CONFIGURE;
-    if (isIndexReady !== this.state.isIndexReady) {
-      this.setState({ isIndexReady });
-      this.props.onIndexReady(isIndexReady);
+  _import = async () => {
+    if (!this._geojsonImporter) {
+      return;
     }
-  };
 
-  _import = async () => {
     //
     // create index
     //
-    const settings = {
+    const settings = ({
       number_of_shards: 1,
-    };
+    } as unknown) as Settings;
```

Comment:
This might be worth creating a new type for. I think it's miscategorized (not just here) as `Settings` in the first place.

Reply:
Ya, this should use another look. I will break this action out into a separate PR because it may get a little tricky since it will effect the route end point typings and will have cascading effects